src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2624,7 +2624,7 @@ private void EmitDefaultValue(TypeSymbol type, bool used, SyntaxNode syntaxNode)
             if (used)
             {
                 // default type parameter values must be emitted as 'initobj' regardless of constraints
-                if (!type.IsTypeParameter())
+                if (!type.IsTypeParameter() && type.SpecialType != SpecialType.System_Decimal)
```

Comment:
The Decimal is a constant, but not an IL constant. From IL prospective it is just a struct, so must be emitted via initobj.
Normally this codepath would not see Decimal, because they would be rewritten into bound literal/constants and codegen knows how to deal with decimal there.

New code that emits Throw expressions emit, however, appears to call EmitDefaultValue directly to put an unreachable emit of a default value as a cute way to balance stack.

By itself that is ok, but it calls this method with any kind of type - IL constants or not, thus causing this issue.